Documented resuming changefeed from Failed state (#15555)
pingcap/docs: ticdc/ticdc-changefeed-overview.md

- `Stopped`: The replication task is stopped, because the user manually pauses the changefeed. The changefeed in this state blocks GC operations.
- `Warning`: The replication task returns an error. The replication cannot continue due to some recoverable errors. The changefeed in this state keeps trying to resume until the state transfers to `Normal`. The maximum retry time is 30 minutes. If it exceeds this time, the changefeed enters a failed state. The changefeed in this state blocks GC operations.
- `Finished`: The replication task is finished and has reached the preset `TargetTs`. The changefeed in this state does not block GC operations.

> - `Failed`: The replication task fails. The changefeed in this state will not keep trying to resume.  To give you enough time to handle the failure, the changefeed in this state blocks GC operations. The duration of the blockage is specified by the `gc-ttl` parameter, with a default value of 24 hours. If the underlying issue is resolved within this duration then the changefeed may be resumed. Otherwise, if the changefeed has been in this state for longer than the `gc-ttl` the replication task cannot resume and cannot be recovered.

Reviewer: ```suggestion
- `Failed`: The replication task fails. The changefeed in this state does not keep trying to resume. To give you enough time to handle the failure, the changefeed in this state blocks GC operations. The duration of the blockage is specified by the `gc-ttl` parameter, with a default value of 24 hours. If the underlying issue is resolved within this duration, you can manually resume the changefeed. Otherwise, if the changefeed remains in this state beyond the `gc-ttl` duration, the replication task cannot resume and cannot be recovered.
```

- ⑥ The changefeed encounters an unrecoverable error and directly enters the failed state. At this time, the changefeed continues to block upstream GC for a duration specified by `gc-ttl`.
- ⑦ The replication progress of the changefeed reaches the value set by `target-ts`, and the replication is completed.

> - ⑧ The changefeed has been suspended for a duration longer than the value specified by `gc-ttl`, thus encountering GC advancement errors, and cannot be resumed.

Reviewer: Please update line 22 as well, which indicates the failed task cannot be resumed.

![image](https://github.com/pingcap/docs/assets/79440533/63bfc0db-071c-41cb-9c72-2216d01dbc03)

Author: Thank you for the comment @qiancai
The change in https://github.com/pingcap/docs/pull/15555/commits/66e7ea3cd45c0a377f40370fbf40b1e14847066a should address this feedback
